packages/core/src/router.ts

```diff
@@ -39,3 +39,29 @@ export function newRouterExecuteTransactionRequest(
 
   return { to: getContractAddress(chainId, 'Router'), data, value };
 }
+
+export interface NewRouterExecuteWithSignerFeeTransactionOptions {
+  chainId: number;
+  routerBatchLogics: IParam.LogicBatchStruct;
+  signer: string;
+  signature: BytesLike;
+  tokensReturn?: string[];
+  value?: BigNumberish;
+  referralCode?: number;
+}
+
+export function newRouterExecuteWithSignerFeeTransactionRequest(
+  options: NewRouterExecuteWithSignerFeeTransactionOptions
+): common.TransactionRequest {
+  const { chainId, routerBatchLogics, signer, signature, tokensReturn = [], value = 0, referralCode = 0 } = options;
+  const iface = Router__factory.createInterface();
+  const data = iface.encodeFunctionData('executeWithSignerFee', [
+    routerBatchLogics,
+    signer,
```

Comment:
Do we need to check the length of `signer` string before encoding it? (Applied to `tokensReturn` string array too)

Reply:
If we can use an `address` type here, the check would be unnecessary.

Reply:
I'm not sure if they are necessary. We can definitely check those including the length of signature. However the calldata will eventually be rejected when the call to contract is performed.

Reply:
Okay. I wondered if there is any chance of a hash collision when the length is unexpected. But for this case, developers can provide any data. Utilizing the length to make a hash collision would be too much work.